src/camp/generators/lib/campch.py

```diff
@@ -150,6 +151,18 @@ def as_text(ari:ari.ARI) -> str:
         enc.encode(ari, buf)
         return buf.getvalue()
 
+    def as_timeperiod(value:datetime) -> str:
+      seconds = value.replace(tzinfo=timezone.utc).timestamp()
+      tv_sec = int(seconds)
+      tv_nsec = int(str(seconds).split(".")[1]) if "." in str(seconds) else 0 
+      return f"{{{tv_sec}, {tv_nsec}}}"
+
+    def as_timedelta(value:timedelta) -> str:
+      seconds = value.total_seconds()
+      tv_sec = int(seconds)
+      tv_nsec = int(str(seconds).split(".")[1]) if "." in str(seconds) else 0 
+      return f"{{{tv_sec}, {tv_nsec}}}"
```

Comment:
Same comment as above.

Reply:
Updated the logic to properly convert fractional seconds to nanoseconds.